python/adaptivity.py.in

```diff
@@ -333,7 +333,65 @@ def detect_colinearity(mesh, tol = 30):
 
     return bfaces_func
 
-
+def adapt_boundary_regions(original_mesh, adapted_mesh, original_regions):
```

Comment:
@croci From what you describe it looks like a bug, but I'm not sure because I'm not sure what is going on on your pictures. 
Can you please open an issue with a "minimal failing example" so I can have a proper look ? If you manage to reproduce this behaviour with an analytic metric that would be awesome, so I don't have to install/use dolfin at first ;)